src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/HardwareIntrinsicHelpers.Aot.cs

```diff
@@ -97,149 +101,96 @@ private static class XArchIntrinsicConstants
+
+            public static int FromInstructionSet(InstructionSet instructionSet)
             {
-                return id switch
+                return instructionSet switch
                 {
-                    "ArmBase" => ArmBase,
-                    "ArmBase_Arm64" => ArmBase_Arm64,
-                    "AdvSimd" => AdvSimd,
-                    "AdvSimd_Arm64" => AdvSimd_Arm64,
-                    "Aes" => Aes,
-                    "Crc32" => Crc32,
-                    "Crc32_Arm64" => Crc32_Arm64,
-                    "Sha1" => Sha1,
-                    "Sha256" => Sha256,
-                    "Atomics" => Atomics,
-                    "Vector64" => Vector64,
-                    "Vector128" => Vector128,
-                    "Rcpc" => Rcpc,
-               _ => throw new NotSupportedException(),
+                    InstructionSet.ARM64_AdvSimd => AdvSimd,
+                    InstructionSet.ARM64_AdvSimd_Arm64 => AdvSimd,
+                    InstructionSet.ARM64_Aes => Aes,
+                    InstructionSet.ARM64_Aes_Arm64 => Aes,
+                    InstructionSet.ARM64_Crc32 => Crc32,
+                    InstructionSet.ARM64_Crc32_Arm64 => Crc32,
+                    InstructionSet.ARM64_Dp => Dp,
+                    InstructionSet.ARM64_Dp_Arm64 => Dp,
+                    InstructionSet.ARM64_Rdm => Rdm,
+                    InstructionSet.ARM64_Rdm_Arm64 => Rdm,
+                    InstructionSet.ARM64_Sha1 => Sha1,
+                    InstructionSet.ARM64_Sha1_Arm64 => Sha1,
+                    InstructionSet.ARM64_Sha256 => Sha256,
+                    InstructionSet.ARM64_Sha256_Arm64 => Sha256,
+                    InstructionSet.ARM64_Atomics => Atomics,
+                    InstructionSet.ARM64_Rcpc => Rcpc,
+
+                    InstructionSet.ARM64_ArmBase => 0,
+                    InstructionSet.ARM64_ArmBase_Arm64 => 0,
+
+                    _ => throw new NotSupportedException(((InstructionSet_ARM64)instructionSet).ToString())
                 };
             }
-
-            public static int FromInstructionSetFlags(InstructionSetFlags instructionSets)
-            {
-                int result = 0;
-
-                foreach (InstructionSet instructionSet in instructionSets)
-                {
-                    result |= instructionSet switch
-                    {
-                        InstructionSet.ARM64_ArmBase => ArmBase,
-                        InstructionSet.ARM64_ArmBase_Arm64 => ArmBase_Arm64,
-                        InstructionSet.ARM64_AdvSimd => AdvSimd,
-                        InstructionSet.ARM64_AdvSimd_Arm64 => AdvSimd_Arm64,
-                        InstructionSet.ARM64_Aes => Aes,
-                        InstructionSet.ARM64_Crc32 => Crc32,
-                        InstructionSet.ARM64_Crc32_Arm64 => Crc32_Arm64,
-                        InstructionSet.ARM64_Sha1 => Sha1,
-                        InstructionSet.ARM64_Sha256 => Sha256,
```

Comment:
Missing cases for `ARM64_Sha1_Arm64` and `ARM64_Sha256_Arm64` here were the direct cause of the crash from #68132